From 646f2cac9e7858d571f27223fc3d2b6a7894e50c Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Fri, 5 Jul 2024 11:00:17 +0200
Subject: [PATCH 1005/1695] dts: stm32: add pll1 configuration for 1.5GHz on
 stm32mp215f-dk

Describe the RCC configuration associated to the CPU operation point
at 1.5GHz.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 1e1a7fd75b6d6daf1a03ad4004d657ccd7a9b8cb dts: stm32: add pll1 configuration for 1.5GHz on stm32mp215f-dk]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I16aede7fcc836e31e6e3ceebde76c55b657de279
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/391659
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-rcc.dtsi | 5 +++++
 core/arch/arm/dts/stm32mp215f-dk.dts                | 6 +++++-
 2 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-rcc.dtsi b/core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-rcc.dtsi
index aabb31c92..6b36a8e0a 100644
--- a/core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-rcc.dtsi
+++ b/core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-rcc.dtsi
@@ -116,6 +116,11 @@
 			cfg = <30 1 1 1>;
 			src = <MUX_CFG(MUX_MUXSEL5, MUXSEL_HSE)>;
 		};
+
+		pll1_cfg_1500Mhz: pll1-cfg-1500Mhz {
+			cfg = <75 2 1 1>;
+			src = <MUX_CFG(MUX_MUXSEL5, MUXSEL_HSE)>;
+		};
 	};
 
 	pll2: st,pll-2 {
diff --git a/core/arch/arm/dts/stm32mp215f-dk.dts b/core/arch/arm/dts/stm32mp215f-dk.dts
index 15b934535..2f0da160e 100644
--- a/core/arch/arm/dts/stm32mp215f-dk.dts
+++ b/core/arch/arm/dts/stm32mp215f-dk.dts
@@ -304,9 +304,13 @@
 		st,ck_cpu1 {
 			cfg_1 {
 				hz = <1200000000>;
-				st,clksrc = <0>;
 				st,pll = <&pll1_cfg_1200Mhz>;
 			};
+
+			cfg_2 {
+				hz = <1500000000>;
+				st,pll = <&pll1_cfg_1500Mhz>;
+			};
 		};
 	};
 };
-- 
2.47.2

